src/melee.cpp

```diff
@@ -1340,6 +1340,33 @@ bool player::can_weapon_block() const
             weapon.has_technique( WBLOCK_3 ));
 }
 
+int blocking_ability( const item &shield )
+{
+    int block_bonus = 2;
+    if (shield.has_technique( WBLOCK_3 )) {
+        block_bonus = 10;
+    } else if (shield.has_technique( WBLOCK_2 )) {
+        block_bonus = 6;
+    } else if (shield.has_technique( WBLOCK_1 )) {
+        block_bonus = 4;
+    }
+    return block_bonus;
+}
+
+std::string player::best_shield() const
```

Comment:
This function doesn't return a string. String is text. A reference to an item isn't a string of text, it's `item &`.
```
item &player::best_shield()
```